### ISSUE TITLE
Sort out Date and Time literals with time-zone and nanoseconds

### ISSUE THREAD

**Issue by wdanilo** [enso-org/enso#5268]
This task is automatically imported from the old Task Issue Board and it was originally created by James Dunkerley.
Original issue is [here](https://www.pivotaltracker.com/story/show/183734954).
<hr />

Current implementation does not serialise the fractional part of seconds when create a constant column within the database.

Look at what is possible with JDBC to allow better serialisation rather than fixed format used currently.

**Comment by jdunkerley** [comment]
Need to research if there is a standard SQL format for datetimes with nanoseconds and time-zone (TZ support is exceeding limited).

**Comment by jdunkerley** [comment]
Will need a custom `set_statement_values` for each DB system.

**Comment by radeusgd** [comment]
We can store as much information as the Database permits.

Currently we do not support dates in SQLite as it is lacking a specific type for these. We may try to add something later by faking the types just like we did for Booleans there, but it is not as high priority.

In Postgres, in #6153 we are introducing microsecond precision date/times - that is the best we can do as the DB is storing dates with microsecond precision - so we just have to lose nanoseconds if we want to use the standard storage types Postgres provide (and we want for compatibility and simplicity).
Similarly, Postgres stores all timestamps in UTC, so we cannot store what timezone was associated with each timestamp. The best we can do is to ensure that the timestamp, once stored in the Database and then retrieved corresponds to the _exact same point in time_ as the one we started with - so the timezone may change, but the point in time will remain exact. Again, that is the best that we can do storing the data as a Postgres date/time type. We could store them in a different format to be able to have additional metadata, but that then would not be compatible with other applications. It is currently not worth it.

So this issue may be closed with #6153.
